### ISSUE TITLE
Problem optimisation with *fix!'ed* variables does not work

### ISSUE THREAD

**Issue by boborbt** [JuliaOpt/Convex.jl#166]
## Problem description

I am encountering a strange behaviour with fixing variables in a problem. In particular in my particular formulation the non-fixed variables do not get optimised.

Here it follows an example of the problem. For the sake of problem description I start by providing you the bare-bone problem. In next section I give the full code to reproduce the problem.

``` julia
α = Variable(M)
α.value = ones(M)  # we set the values just to check if they change

β = Variable(K)
β.value = [1,-1,1,-1, ]  # we set the values just to check if they change

# the following could be simplified as X * (P .* α) * β, the current
# formulation is a workaround for a bug in the .* operator.
# note: P is an M x K constant matrix
model = P .* (α * ones(1,K)) * β

# X is an N x M data matrix and y is the N x 1 target vector 
loss = norm( X * model - y)
fix!(α)
problem = minimize(loss)
solve!(problem)
```

the problem is that at the end of the computation the β variables are not optimised (the loss is the same as it were before solving the problem and the value of β  is unchanged). The problem is apparently tied to some interaction with the α variables. In fact,  if I substitute the α with a constant, the β does get optimized (I just redefined the model as: `model = P .* (α.value * ones(1,K)) * β` and apparently everything works).

I am new to julia and Convex so there might be something I am missing, but this does looks like a bug to me. Am I right? Is there any workaround (aside from substituting the α with constants?

Thanks in advance for any help.
## Complete code to reproduce the problem

``` julia

using Convex

# sample P matrix
P = [
  [1 0 0 0]
  [1 0 0 0]
  [1 0 0 0]
  [0 1 0 0]
  [0 1 0 0]
  [0 1 0 0]
  [0 0 1 0]
  [0 0 1 0]
  [0 0 0 1]
  [0 0 0 1]
  [0 0 0 1]
]

M,K = size(P)
N = 50

# generating parameters to compute the target model
a = rand(M)
b = rand(K)

# generating the data for the example
X = rand(N,M)
y = X * (P .* (a * ones(1,K))) * b

# formulating the problem
α = ones(M)
α = Variable(M)
α.value = ones(M) 

β = Variable(K)
β.value = [1,-1,1,-1, ]

model = P .* (α * ones(1,K)) * β
loss = norm( X * model - y)

# solving the problem and checking the results
println("Before running $(evaluate(loss))")
println(β.value)

fix!(α)
problem = minimize(loss)
solve!(problem)

println(β.value)
println(problem.optval)
```
## Output of the sample program

Here is the complete output I observe:

``` bash
WARNING: Method definition norm(Convex.AbstractExpr, Symbol) in module Convex at /[...]/.julia/v0.5/Convex/src/atoms/second_order_cone/norm.jl:45 overwritten at deprecated.jl:49.
Before running 22.434070163416827
[1,-1,1,-1]
----------------------------------------------------------------------------
    SCS v1.1.8 - Splitting Conic Solver
    (c) Brendan O'Donoghue, Stanford University, 2012-2015
----------------------------------------------------------------------------
Lin-sys: sparse-direct, nnz in A = 3
eps = 1.00e-04, alpha = 1.80, max_iters = 20000, normalize = 1, scale = 5.00
Variables n = 2, constraints m = 52
Cones:  primal zero / dual free vars: 1
    soc vars: 51, soc blks: 1
Setup time: 6.77e-04s
----------------------------------------------------------------------------
 Iter | pri res | dua res | rel gap | pri obj | dua obj | kap/tau | time (s)
----------------------------------------------------------------------------
     0|      inf       inf       nan      -inf       inf       inf  4.46e-04
   100| 1.01e-04  5.00e-01  9.60e-01  2.42e+01 -0.00e+00  0.00e+00  5.43e-04
   200| 9.93e-05  5.00e-01  9.59e-01  2.33e+01 -0.00e+00  0.00e+00  6.36e-04
   300| 3.47e-04  1.49e-01  1.27e-01  2.24e+01  2.91e+01  0.00e+00  7.30e-04
   340| 5.43e-08  2.65e-05  2.59e-05  2.24e+01  2.24e+01  0.00e+00  7.72e-04
----------------------------------------------------------------------------
Status: Solved
Timing: Solve time: 7.98e-04s
    Lin-sys: nnz in L factor: 57, avg solve time: 6.90e-07s
    Cones: avg projection time: 8.45e-08s
----------------------------------------------------------------------------
Error metrics:
dist(s, K) = 1.1233e-15, dist(y, K*) = 0.0000e+00, s'y/m = 1.0035e-16
|Ax + s - b|_2 / (1 + |b|_2) = 5.4319e-08
|A'y + c|_2 / (1 + |c|_2) = 2.6480e-05
|c'x + b'y| / (1 + |c'x| + |b'y|) = 2.5943e-05
----------------------------------------------------------------------------
c'x = 22.4341, -b'y = 22.4353
============================================================================
[1,-1,1,-1]
22.434068275886943
```
## Output of the sample program when α are substituted by constants

If I substitute the model as it follows:

``` julia
model = P .* (α.value * ones(1,K)) * β
```

I get the following output

``` bash
WARNING: Method definition norm(Convex.AbstractExpr, Symbol) in module Convex at /[...]/.julia/v0.5/Convex/src/atoms/second_order_cone/norm.jl:45 overwritten at deprecated.jl:49.
Before running 12.953606550748447
[1,-1,1,-1]
----------------------------------------------------------------------------
    SCS v1.1.8 - Splitting Conic Solver
    (c) Brendan O'Donoghue, Stanford University, 2012-2015
----------------------------------------------------------------------------
Lin-sys: sparse-direct, nnz in A = 203
eps = 1.00e-04, alpha = 1.80, max_iters = 20000, normalize = 1, scale = 5.00
Variables n = 6, constraints m = 52
Cones:  primal zero / dual free vars: 1
    soc vars: 51, soc blks: 1
Setup time: 7.08e-04s
----------------------------------------------------------------------------
 Iter | pri res | dua res | rel gap | pri obj | dua obj | kap/tau | time (s)
----------------------------------------------------------------------------
     0| 1.59e+00  1.99e+00  8.16e-01 -3.25e+00  1.20e+00  1.21e-15  3.40e-03
    80| 2.12e-06  9.08e-07  6.13e-06  4.79e-01  4.79e-01  1.30e-16  3.53e-03
----------------------------------------------------------------------------
Status: Solved
Timing: Solve time: 3.57e-03s
    Lin-sys: nnz in L factor: 273, avg solve time: 2.66e-06s
    Cones: avg projection time: 1.04e-07s
----------------------------------------------------------------------------
Error metrics:
dist(s, K) = 1.4729e-17, dist(y, K*) = 0.0000e+00, s'y/m = 3.6946e-18
|Ax + s - b|_2 / (1 + |b|_2) = 2.1161e-06
|A'y + c|_2 / (1 + |c|_2) = 9.0765e-07
|c'x + b'y| / (1 + |c'x| + |b'y|) = 6.1286e-06
----------------------------------------------------------------------------
c'x = 0.4786, -b'y = 0.4786
============================================================================
[0.127554; 0.19955; 0.0778711; 0.279063]
0.4786006438371289
```
